expo/expo
# [home] Updated permissions and fixed format

# Why

Home is messy, unified permissions system.

# How

* Added flow to prompt the user before asking for permission
* Added prompt to redirect to app settings when the permission is denied
* Sorted imports
* Converted some components to PureComponents to cut down on re-renders
* Removed AppState observer for permission as the app restarts on permission change

# Test Plan

* Run on iOS and Android
* open QR -> deny first permission prompt -> repeat (should work)
* open QR -> accept first permission prompt -> reject second -> third should open the settings app to the Expo app page.
